src/tests/roc_fec/target_openfec/test_writer_reader.cpp

```diff
@@ -902,5 +902,122 @@ TEST(writer_reader, repaired_block_numbering) {
     }
 }
 
+TEST(writer_reader, resize_souce_block_increase_decrease) {
+    OFEncoder encoder(config, FECPayloadSize, allocator);
+    OFDecoder decoder(config, FECPayloadSize, buffer_pool, allocator);
+
+    CHECK(decoder.valid());
+    CHECK(encoder.valid());
+
+    PacketDispatcher dispatcher;
+
+    Writer writer(config, FECPayloadSize, encoder, dispatcher, source_composer,
+                  repair_composer, packet_pool, buffer_pool, allocator);
+    Reader reader(config, decoder, dispatcher.source_reader(), dispatcher.repair_reader(),
+                  rtp_parser, packet_pool, allocator);
+
+    CHECK(reader.valid());
+    CHECK(writer.valid());
+
+    const size_t block_sizes[] = { 20, 25, 35, 43, 33, 23, 13 };
+    CHECK(block_sizes[0] == NumSourcePackets);
+
+    packet::seqnum_t pack_n = 0;
+
+    for (size_t n = 0; n < ROC_ARRAY_SIZE(block_sizes) - 1; ++n) {
+        writer.resize(block_sizes[n + 1]);
```

Comment:
Because we set the size of the next block.

Reply:
Will the following comment help you understand `n + 1`?

```
We set the size of the source block before writing any packets into Writer. As a result, the updated source block size will be used only on the next block.
```

Reply:
> As a result the updated source block size will be used only on the next block.

Why is this so even if we call writer.resize *before* writing the block? I would expect that if the resize() is called before writing the first packet in a block, it will take effect starting from *this* block.

Reply:
You are right but it was implemented in a slightly different way. resize just promises you that next block will have the provided size. I guess that we need also to handle the case when resize is called at the beginning of the block.

Reply:
I have reworked. FYI: https://github.com/roc-project/roc/pull/174#discussion_r283100657